proposals/0010-resource-element-type-validation.md

```diff
@@ -0,0 +1,223 @@
+* Proposal: [0010](0010-resource-element-type-validation.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: Joshua Batista
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+* Issues: [#75676](https://github.com/llvm/llvm-project/issues/75676)
+
+## Introduction
+Resources are often used in HLSL, with various resource element types (RETs).

```

Comment:
I'm not a fan of the new acronym "RET", since it collides so readily with an instruction for return from procedure.

I think I'd prefer "resource element type" or "element type" or "resource element", or just "element" for short where the context is sufficient.

Reply:
One thought is to use the term "typed resource element" (maybe "TRE" for short, but I'm not sure it's necessary).  That would be a specific thing with specific constraints, avoiding the collision with the `StructuredBuffer` element type or `ByteAddressBuffer` templated `Load`/`Store` element type where the constraints are different.

Reply:
I see how typed resource element is more specific, though I'd like to address element types in this spec specifically. How does TRET (typed resource element types) sound? There are many references to this concept so I think an abbreviation is worthwhile.

Reply:
I'm not a fan of the RET acronym either. I agree with Tex that in many places it can be shortened to "element type", "resource element", or just "element" or "type".

Reply:
Ok, I'll remove acronyms.

Reply:
Thanks Josh! In my opinion it looks much cleaner now.